packages/serverpod/lib/src/server/websocket_request_handlers/method_websocket_request_handler.dart

```diff
@@ -276,31 +513,21 @@ class _MethodStreamManager {
         );
       }
 
-      _postMessage(
-        endpoint: message.endpoint,
-        method: message.method,
-        connectionId: message.connectionId,
-        message: CloseMethodStreamCommand.buildMessage(
-          endpoint: message.endpoint,
-          connectionId: message.connectionId,
-          method: message.method,
-          reason: CloseReason.error,
-        ),
-      );
-
       await session.close(error: e, stackTrace: stackTrace);
-      await closeStream(
+      await _closeMethodStream(
         endpoint: message.endpoint,
         method: message.method,
         connectionId: message.connectionId,
+        reason: CloseReason.error,
       );
       return;
     }
 
     // TODO: Support nullable return types.
-    // Becuase encodeWithType doens't support nullable we can't encode null
+    // Because encodeWithType does not support nullable we can't encode null
     // values.
-    if (methodConnector.returnsVoid == false && result != null) {
+    if (methodConnector.returnType != MethodStreamReturnType.voidType &&
+        result != null) {
```

Comment:
Nice, that would probably have been a neater solution! But Klaudio has actually added support for null values so I can probably remove this completely in another PR. Then I can also add tests for `null` return values.

Reply:
Oh wait, I misunderstood your suggestion. We cant just set it to null because then we won't have that field Serialized with type. We now have support for null values following our special serialization. But I will add that in a separate PR since I would like to add tests for that case at the same time.